thinknimble/tn-spa-bootstrapper
# feat: allow to send templates preview through email

## What this does

#193 introduced an endpoint to preview how templates would look like.
This PR adds a small form to be able to actually email those previews as well.

## Checklist


## How to test

1. Create a new project based off this branch: `cookiecutter git@github.com:thinknimble/tn-spa-bootstrapper.git -checkout feat/allow-to-send-templates-preview-through-email`
2. Run `pipenv install` and `docker compose up server`
3. Try previewing a template (i.e.: in your browser): `http://localhost:8000/api/template_preview/?template=registration/password_reset.html&site_url=http://localhost:8000/&support_email=franco.pelusso@thinknimble.com`
** to test this [PR](https://tn-spa-bootstrapper-pr-247.herokuapp.com/api/template_preview/?template=registration/password_reset.html&site_url=http://localhost:8000/&support_email=franco.pelusso@thinknimble.com)**
4. Try out filling your email in the input and sending.
5. You should see the full HTML email in the logs.
6. If you configure `MAILGUN`, you should be able to actually receive the email.
7. You can also trigger sending the email by doing a `POST` request directly to that same endpoint, providing all parameters + `"_send_to": "franco.pelusso@thinknimble.com"`
